hack/cmd_util.sh

```diff
@@ -147,41 +147,50 @@ os_cmd_internal_tmperr="${os_cmd_internal_tmpdir}/tmp_stderr.log"
 # any error exiting settings or traps set by upstream callers by masking the return code of the command 
 # with the return code of setting the result variable on failure.
 function os::cmd::internal::expect_exit_code_run_grep() {
-	cmd=$1
+	local cmd=$1
 	# default expected cmd code to 0 for success
-	cmd_eval_func=${2:-os::cmd::internal::success_func}
+	local cmd_eval_func=${2:-os::cmd::internal::success_func}
 	# default to nothing 
-	grep_args=${3:-} 
+	local grep_args=${3:-} 
 	# default expected test code to 0 for success
-	test_eval_func=${4:-os::cmd::internal::success_func}
+	local test_eval_func=${4:-os::cmd::internal::success_func}
 
 	os::cmd::internal::init_tempdir
 
-	echo "Running  ${cmd}..."
-	
-	cmd_result=$( os::cmd::internal::run_collecting_output "${cmd}"; echo $? )
-	cmd_succeeded=$( ${cmd_eval_func} "${cmd_result}"; echo $? )
+	local name=$(os::cmd::internal::describe_call "${cmd}" "${cmd_eval_func}" "${grep_args}" "${test_eval_func}")
+	echo "${name}..."
+
+	local start_time=$(os::cmd::internal::seconds_since_epoch)
 
-	test_result=0
+	local cmd_result=$( os::cmd::internal::run_collecting_output "${cmd}"; echo $? )
+	local cmd_succeeded=$( ${cmd_eval_func} "${cmd_result}"; echo $? )
+
+	local test_result=0
 	if [[ -n "${grep_args}" ]]; then
 		test_result=$( os::cmd::internal::run_collecting_output 'os::cmd::internal::get_results | grep -Eq "${grep_args}"'; echo $? )
 		
 	fi
-	test_succeeded=$( ${test_eval_func} "${test_result}"; echo $? )
+	local test_succeeded=$( ${test_eval_func} "${test_result}"; echo $? )
 
-	os::text::clear_last_line
+	local end_time=$(os::cmd::internal::seconds_since_epoch)
+	local time_elapsed=$(echo "scale=9; ${end_time} - ${start_time}" | bc | xargs printf '%11.9f') # we need leading zeroes for parsing later
```

Comment:
> Add a comment saying what the intended output format is. Since you said something about parsing, I'm guessing millis?

Wait, floating point? kids these days.